core/src/main/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriter.java

```diff
@@ -122,6 +123,8 @@ public UnsafeShuffleWriter(
     this.taskContext = taskContext;
     this.sparkConf = sparkConf;
     this.transferToEnabled = sparkConf.getBoolean("spark.file.transferTo", true);
+    this.initialSortBufferSize = sparkConf.getInt("spark.shuffle.initial.buffer.size",
```

Comment:
@srowen - Making the size significantly larger (in our case we are using 64MB instead of 64KB by default) will not work well in case you have small amount of heap memory available. Also that might not be helpful for small datasets.